src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -186,20 +187,34 @@ private OkObjectResult FormatFindResult(JsonDocument jsonDoc, FindRequestContext
         {
             JsonElement jsonElement = jsonDoc.RootElement.Clone();
 
+            // When there are no rows returned from the database, the jsonElement will be an empty array.
+            // In that case, the response is returned as is.
+            if (jsonElement.ValueKind is JsonValueKind.Array && jsonElement.EnumerateArray().Count() == 0)
```

Comment:
instead of `jsonElement.EnumerateArray().Count() == 0` could you do something like `!jsonElement.Any()`? if the count >0 then you're enumerating the whole array unnecessarily.

Reply:
Good catch. I've updated to use `jsonElement.GetArrayLength() == 0`